zubhub_frontend/zubhub/src/hooks/useIsInViewPort.hook.js

```diff
@@ -0,0 +1,24 @@
+import { useState, useMemo, useEffect } from "react";
+
+
+export function useIsInViewport(ref) {
```

Comment:
this hook generally watches over the reference element `ref` using the `IntersectionObserver` API. This will be used to check if an element is in the `view port`

Reply:
Can you move the IntersectionObserver creation to the useEffect as calling disconnect will stop emitting any new events?

Usually ref won't change, the value ref.current changes.  Just curious how the useEffect is working if the target element is remounted or re-rendered.

Reply:
Any updates on this @coderatomy ?

Reply:
Sorry for the late reply @tuxology @chaitanya-cloudknox 
Ideally, the useEffect hook handles remounting or re-rendering of the target element by cleaning up the previous IntersectionObserver and creating a new one to observe the updated element, thanks to the dependency array [ref]. This ensures that the hook stays in sync with the current state of the DOM.